modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java

```diff
@@ -243,7 +243,13 @@ protected static void filterAppDisplays(
 		while (iterator.hasNext()) {
 			AppDisplay appDisplay = iterator.next();
 
-			if ((state > 0) && (appDisplay.getState() != state)) {
+			List<Bundle> appDisplayBundles = appDisplay.getBundles();
+
+			int appDisplayBundlesSize = appDisplayBundles.size();
+
+			if ((appDisplay.getState() != state) &&
+				(appDisplayBundlesSize == 0)) {
```

Comment:
If the `appDisplayBundlesSize == 0` condition is `true`, shouldn't we always remove the iterator?  Is there a case where the given `appDisplay` doesn't have any bundles but we still want to display it?

Reply:
I could not find such a case, so I believe we should always remove the iterator.
Let me send a new pull request to you where this is fixed.